File: app/src/main/java/com/asmat/rolando/popularmovies/adapters/TrailersLinearAdapter.java
package com.asmat.rolando.popularmovies.adapters;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import com.asmat.rolando.popularmovies.R;
import com.asmat.rolando.popularmovies.models.AdapterOnClickHandler;
import com.asmat.rolando.popularmovies.networking.models.Video;
import com.squareup.picasso.Picasso;

public class TrailersLinearAdapter extends BaseLinearAdapter<Video,TrailersLinearAdapter.ViewHolder> {

    public TrailersLinearAdapter(AdapterOnClickHandler<Video> clickHandler) {
        super(clickHandler);
    }

    @Override
    int getLayoutForLinearItem() {
        return R.layout.trailer_linear_item;
    }

    @Override
    ViewHolder createViewHolder(View view) {
        return new TrailersLinearAdapter.ViewHolder(view);
    }

    class ViewHolder extends BaseLinearAdapter.ViewHolder {
        final ImageView thumbnail;
        final TextView caption;

        ViewHolder(View view) {
            super(view);
            thumbnail = view.findViewById(R.id.iv_trailer_thumbnail);
            caption = view.findViewById(R.id.tv_trailer_caption);
            thumbnail.setOnClickListener(this);
        }

        @Override
        void bind(Object item) {
            if (item instanceof Video) {
                Video video = (Video) item;
                caption.setText(video.getName());
                String url = video.getYouTubeThumbnailURL();
                Picasso.with(thumbnail.getContext())
                        .load(url)
                        .into(thumbnail);
            }
        }
    }
}

File: app/src/main/java/com/asmat/rolando/popularmovies/ui/adapters/linear/TrailersLinearAdapter.kt
package com.asmat.rolando.popularmovies.ui.adapters.linear

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.asmat.rolando.popularmovies.R
import com.asmat.rolando.popularmovies.networking.the.movie.db.models.VideosResponse
import com.asmat.rolando.popularmovies.utilities.URLUtils
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.trailer_linear_item.view.*

private typealias Video = VideosResponse.Video
private typealias VideoViewHolder = TrailersLinearAdapter.ViewHolder

class TrailersLinearAdapter(clickHandler: (Video) -> Unit?) :
        BaseLinearAdapter<Video, VideoViewHolder>(clickHandler) {

    override val layoutForLinearItem: Int
        get() = R.layout.trailer_linear_item


    override fun createViewHolder(view: View): ViewHolder {
        return ViewHolder(view)
    }

    inner class ViewHolder(view: View) : BaseLinearAdapter<Video, VideoViewHolder>.ViewHolder(view) {
        private val thumbnail: ImageView = view.iv_trailer_thumbnail
        private val caption: TextView = view.tv_trailer_caption

        init {
            thumbnail.setOnClickListener(this)
        }

        override fun bind(item: Video) {
            val url = URLUtils.getYoutubeThumbnailURL(item.key)
            Picasso.with(thumbnail.context)
                    .load(url)
                    .into(thumbnail)
            caption.text = item.name
        }
    }
}